lingpy/convert/strings.py

```diff
@@ -404,40 +401,69 @@ def multistate2nex(taxa, matrix, filename='', missing="?"):
     return
 
 
+def nexus_slug(s):
+    """
+    Converts a string to a nexus "safe" representation (i.e. removes
+    many unicode characters and removes some punctuation characters).
+
+    Parameters
+    ----------
+    s : str
+        A string to convert to a nexus safe format.
+
+    Returns
+    -------
+    s : str
+        A string containing a nexus safe label.
+
+    """
+    s = unicodedata.normalize('NFKD', s)
+    s = "".join([c for c in s if not unicodedata.combining(c)])
```

Comment:
If you replace this by:
```python
s = "".join([c for c in s if ord(c)<128 and c not in "();?.\":[]/!"])
```
you can get rid of the next line, plus, Chinese character will deleted. I'd also suggest to finally replace all spaces by an underscore, and to issue a warning if nothing is left, or even better, just raise an exception.

Reply:
Note that exclamation mark was also missing in the list of forbidden things.

Reply:
But in general, I just discussed with @xrotwang to  use clldutils-slug which allows to leave spaces via option already, and simply add another option to leave capitalization, which is probably useful.

Reply:
Addon, clldutils.misc.slug already offers all we need:
```python
>>> slug('Äther Ist däi (bla)', lowercase=False, remove_whitespace=False)
'Ather Ist dai bla'
```

Reply:
clldutils.misc.slug removes useful punctuation like `_`